cose/key.go

```diff
@@ -95,8 +95,10 @@ func NewCoseCommonKey(coseKey map[int64]interface{}) (*CoseCommonKey, error) {
 	}
 
 	kid := coseKey[KeyIDLabel]
-	// TODO: error handling
-	kidBytes, _ := kid.([]byte)
+	kidString, ok := kid.(string)
```

Comment:
This did not work for me with []byte - Even when I set the header with []byte this didn't seem to work. Are we okay with this change @robinbryce ? It returns []byte from the getter method to conform to the interface

Reply:
I believe we are, in golang strings are not implicitly handles as unicode I think. So essentially string is bytes once we are serialized.